### DISCUSSION TITLE
Kademlia conflict resolution?

### DISCUSSION THREAD

**Discussion by laptou** [libp2p/rust-libp2p#2415] [Q&A] [answered]
How does Kademlia handle conflict resolution? For example, if two remote nodes update the same DHT entry at the same time, how is it decided whose value will "win"? Is there currently a way to merge the values if a conflict is detected?

**Comment by mxinden** [comment, marked as answer]
> how is it decided whose value will "win"?

This is up to the application.

A user can set [`KademliaStoreInserts::FilterBoth`](https://github.com/libp2p/rust-libp2p/blob/3749dd56f16590c7a312eb6f1d51195f59e34817/protocols/kad/src/behaviour.rs#L150-L159) and thus decide whether the existing or the new value should "win".

Let me know if the above is helpful.

**Reply by laptou** [reply]
This is perfect, thank you!
